datastore/src/main/java/com/ds/datastore/BookStoreController.java

```diff
@@ -144,10 +141,13 @@ protected void addServer(@RequestBody String json, @PathVariable String id) {
     }
 
     @GetMapping("/bookstores")
-    protected CollectionModel<EntityModel<BookStore>> all() throws Exception {
+    protected CollectionModel<EntityModel<BookStore>> all(@RequestParam(required = false) List<String> id) throws Exception {
+        if(id != null) {
```

Comment:
Because of our most recent addition, where if a client requests a store or book that doesn't exist, we ignore it and return the the items that do exist,  that won't be so simple

Reply:
but I would still assume you can isolate some of the logic that gets the store and populates a domain entity...

Reply:
Good point. Fixed with the recent commits